internal/engine/image_build_and_deployer.go

```diff
@@ -80,7 +106,7 @@ func (ibd *ImageBuildAndDeployer) BuildAndDeploy(ctx context.Context, st store.R
 		ibd.analytics.Timer("build.image", time.Since(startTime), tags)
 	}()
 
-	numStages := len(iTargets)
+	numStages := len(iTargets) * 2 // each image target has two stages: one for build, and one for push
```

Comment:
not really related to this pr (except that you're updating the logic) but this probably needs to take into account the number of NeedsImageBuild(). i wish we had a better way to assert this number is correct.